Hippo.Web/Services/BulkLoadService.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Hippo.Core.Data;
+using Hippo.Core.Domain;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+using static Hippo.Core.Domain.Account;
+
+namespace Hippo.Web.Services
+{
+    public interface IBulkLoadService
+    {
+        Task<int> Load(Cluster cluster, string data);
+    }
+
+    public class BulkLoadService : IBulkLoadService
+    {
+        private readonly AppDbContext _dbContext;
+        private readonly IIdentityService _identityService;
+
+        public BulkLoadService(AppDbContext dbContext, IIdentityService identityService)
+        {
+            _dbContext = dbContext;
+            _identityService = identityService;
+        }
+
+
+
+        public async Task<int> Load(Cluster cluster, string data)
+        {
+            var count = 0;
+            var pairs = data.Split(",");
+
+            Log.Information($"Starting bulk load for {pairs.Count()} accounts");
+
+            //Create missing users
+            var kerbs = new List<string>();
+            var kerbsInDb = new List<string>();
+            var sponsorKerbs = new List<string>();
+            foreach(var pair in pairs)
+            {
+                var kerbIs = pair.Split("-");
+                kerbs.Add(kerbIs[0].Trim());
+                kerbs.Add(kerbIs[1].Trim());
+                sponsorKerbs.Add(kerbIs[1].Trim()); //Load sponsors
+            }
+            kerbs = kerbs.Distinct().ToList();
+            sponsorKerbs = sponsorKerbs.Distinct().ToList();
+            Log.Information($"Unique Kerbs: {kerbs.Count()}");
+
+            foreach(var kerb in kerbs)
+            {
+                var user = await _identityService.GetByKerberos(kerb);
+                if(user == null)
+                {
+                    Log.Error($"Kerb not found: {kerb}");
+                    continue;
+                }
+                if(!await _dbContext.Users.AnyAsync(a => a.Iam == user.Iam))
+                {
+                    Log.Information($"Adding user: {kerb}");
+                    await _dbContext.Users.AddAsync(user);
+                }
+                kerbsInDb.Add(kerb);
+            }

```

Comment:
Not sure if multiple calls to identity service can be avoided, but multiple calls to db certainly can without getting fancy with udf/temp table stuff if you load all the kerbs into memory (swap Iam for Id if you want to search by that) 
```csharp
var currentKerbs = _dbContext.Users.Select(x => x.Id).ToHashSet();
var kerbsToBeAdded = kerbs.Where(id => !currentKerbs .Contains(id));
```
I wouldn't expect the number of user records to ever get high enough to cause a memory issue.